src/js/components/TaskTable.js

```diff
@@ -39,9 +39,59 @@ class TaskTable extends React.Component {
   }
 
   getStatusValue(task) {
+    let taskHealth = this.getTaskHealth(task);
+    if (taskHealth === true) {
+      return 'Healthy';
+    }
+    if (taskHealth === false) {
+      return 'Unhealthy';
+    }
+
     return TaskStates[task.state].displayName;
   }
 
+  getTaskHealth(task) {
+    let mesosTaskHealth = this.getTaskHealthFromMesos(task);
+    if (mesosTaskHealth !== null) {
+      return mesosTaskHealth;
+    }
+    let marathonTaskHealth = this.getTaskHealthFromMarathon(task);
+    if (marathonTaskHealth !== null) {
+      return marathonTaskHealth;
+    }
+
+    return null;
+  }
+
+  getTaskHealthFromMarathon(task) {
```

Comment:
We could, but my fear is we'll just end up hiding the problem and never fixing it. IMO this shouldn't be done in the component during render at all...